core/main/src/firebolt/handlers/provider_registrar.rs

```diff
@@ -499,6 +495,9 @@ impl ProviderRegistrar {
                                             }
                                         }
                                         return Ok(Value::Object(response_map));
+                                    } else {
+                                        // Method returns a non-object type, just return it.
+                                        return Ok(provider_response_value);
```

Comment:
Previously the OpenRPC Validator would return the name of the result field as an actual field in the result, causing the problem. That's now corrected in the validator, and here if we don't get a list of properties from the schema (because it's a non-object) in order to do app ID injection, we just return the actual response value (in this case a `Value::String`)